examples/vue-instantsearch/src/App.vue

```diff
@@ -0,0 +1,428 @@
+<template>
+  <div>
+    <header class="header">
+      <div class="header-wrapper wrapper">
+        <nav class="header-nav">
+          <a href="/">Home</a>
+        </nav>
+        <div id="autocomplete"></div>
+      </div>
+    </header>
+
+    <ais-instant-search
+      :search-client="searchClient"
+      :index-name="indexName"
+      :routing="routing"
+      :middlewares="middlewares"
+    >
+      <ais-configure
+        :attributes-to-snippet.camel="['name:7', 'description:15']"
+        :snippet-ellipsis-text.camel="'…'"
+      ></ais-configure>
+      <div class="container wrapper">
+        <div>
+          <ais-panel>
+            <template v-slot:header>Categories</template>
+            <ais-hierarchical-menu
+              :attributes="[
+                'hierarchicalCategories.lvl0',
+                'hierarchicalCategories.lvl1',
+              ]"
+            ></ais-hierarchical-menu>
+          </ais-panel>
+        </div>
+        <div>
+          <ais-search-box><template></template></ais-search-box>
+          <ais-hits>
+            <template v-slot:item="{ item }">
+              <article class="hit">
+                <div class="hit-image">
+                  <img :src="item.image" :alt="item.name" />
+                </div>
+                <div>
+                  <h1>
+                    <ais-snippet :hit="item" attribute="name" />
+                  </h1>
+                  <div>
+                    By <strong>{{ item.brand }}</strong> in
+                    <strong>{{ item.categories[0] }}</strong>
+                  </div>
+                </div>
+              </article>
+            </template>
+          </ais-hits>
+          <ais-pagination :show-first="false" :show-last="false" />
+        </div>
+      </div>
+    </ais-instant-search>
+  </div>
+</template>
+
+<script>
+import { h, Fragment, render } from 'vue';
+
+import algoliasearch from 'algoliasearch/lite';
+
+import { history as historyRouter } from 'instantsearch.js/es/lib/routers';
+import { singleIndex as singleIndexMapping } from 'instantsearch.js/es/lib/stateMappings';
+
+import { autocomplete } from '@algolia/autocomplete-js';
+import { createQuerySuggestionsPlugin } from '@algolia/autocomplete-plugin-query-suggestions';
+import { createLocalStorageRecentSearchesPlugin } from '@algolia/autocomplete-plugin-recent-searches';
+
+import '@algolia/autocomplete-theme-classic';
+
+const INSTANT_SEARCH_INDEX_NAME = 'instant_search';
+const INSTANT_SEARCH_HIERARCHICAL_ATTRIBUTE = 'hierarchicalCategories.lvl0';
+
+const searchClient = algoliasearch(
+  'latency',
+  '6be0576ff61c053d5f9a3225e2a90f76'
+);
+
+function autocompleteMiddleware({ instantSearchInstance }) {
```

Comment:
what do you think about adding this middleware in its own file? it does make the entry point less "autocomplete"-focused, but now it's in the middle of the file making it hard to skim what's part of the example and what's part of setting up autocomplete